src/apps/backend/modules/comments/internal/comment-writer.ts

```
@@ -0,0 +1,33 @@
+import { CreateCommentParams, EditCommentParams, Comment } from '../types';
+import CommentRepository from './store/comment-repository';
+import CommentUtil from './comment-util';
+import { Types } from 'mongoose';
+
+export default class CommentWriter {
+  public static async createComment(params: CreateCommentParams): Promise<Comment> {
+    const commentDb = await CommentRepository.create({
+      task: new Types.ObjectId(params.taskId),
+      account: new Types.ObjectId(params.accountId),
+      comment: params.comment,
+      active: true,
+    });
+    return CommentUtil.convertCommentDBToComment(commentDb);
+  }
+
+  public static async editComment(params: EditCommentParams): Promise<Comment | null> {
+    const commentDb = await CommentRepository.findByIdAndUpdate(
+      params.commentId,
+      { comment: params.comment },
+      { new: true }
+    );
+    return commentDb ? CommentUtil.convertCommentDBToComment(commentDb) : null;
```

If we don't find commentDb then we should throw error instead of returning null